super_csv/mixins.py

```diff
@@ -95,7 +95,10 @@ def save(self, op_name=None):
         Save the state of this object to django storage.
         """
         state = self.__dict__.copy()
-        for k, v in state.items():
+        user = state.get('_user')
```

Comment:
I believe the motivation for the whole `_<field>` private attribute set is to make sure we aren't trying to serialize a potentially deeply nested object (e.g. a django `User`). I think it would still make sense to serialize the `user_id`, however to identify which user should be attached to the operation when it gets rehydrated; do you think it would be loads more work to take the approach of checking for a `user_id` & creating a `User` object from that when a serialized operation is loaded, rather than creating an exception to the overarching way in which the `_<field>`s are handled.

Reply:
We're still removing the user from the state, it's not being serialized. 
My original approach was to do what you said, store the user.id in the serialized state, but this seems easier since we need to pass user to record_operation (or somehow get it into the CSVOperation) anyway. 

The CSVOperation not having the user was the cause of https://openedx.atlassian.net/browse/EDUCATOR-4650

Reply:
Yeah, good point.  I think `json` will eventually raise an exception when trying to serialize a Django `User` object during save.  Matt's idea makes sense to me.

Reply:
I also initially misread this as serializing the user.  I think what you currently have in this PR works fine.

Reply:
> we need to pass user to record_operation (or somehow get it into the CSVOperation) anyway

Can we `User.objects.get(id=self.user_id)` in the course of `load`ing though? I would need more words to understand why passing the User to `record_operation` is _necessary_, as opposed to just how it's currently being done.

I could also imagine us changing this by raising an exception if `record_operation` is called without `self._user` set, and moving the responsibility to the call site to ensure that the user's been hydrated appropriately based on the context.

Reply:
So you're suggesting that in save() we save the user in the state by adding a 'user_id' field, and then in load() rather than using the User on the CSVOperation, we use the user that's been saved in user_id?